Zend/zend_weakrefs.c

```diff
@@ -0,0 +1,235 @@
+	zend_throw_error(NULL, "WeakReference objects do not support " thing);
+
+static zval* zend_weakref_no_write(zval *object, zval *member, zval *value, void **rtc) {
+	zend_weakref_unsupported("properties");
+
+	return &EG(uninitialized_zval);
+}
+
+static zval* zend_weakref_no_read(zval *object, zval *member, int type, void **rtc, zval *rv) {
+	if (!EG(exception)) {
+		zend_weakref_unsupported("properties");
+	}
+
+	return &EG(uninitialized_zval);
+}
+
+static zval *zend_weakref_no_read_ptr(zval *object, zval *member, int type, void **rtc) {
+	zend_weakref_unsupported("property references");
+	return NULL;
+}
+
+static int zend_weakref_no_isset(zval *object, zval *member, int hse, void **rtc) {
+	if (hse != 2) {
+		zend_weakref_unsupported("properties");
+	}
+	return 0;
+}
+
+static void zend_weakref_no_unset(zval *object, zval *member, void **rtc) {
+	zend_weakref_unsupported("properties");
+}
+
+ZEND_BEGIN_ARG_INFO_EX(zend_weakref_construct_arginfo, 0, 0, 1)
+	ZEND_ARG_TYPE_INFO(0, referent, IS_OBJECT, 0)
+ZEND_END_ARG_INFO()
+
+ZEND_BEGIN_ARG_WITH_RETURN_TYPE_INFO_EX(zend_weakref_get_arginfo, 0, 0, IS_OBJECT, 1)
+ZEND_END_ARG_INFO()
+
+ZEND_METHOD(WeakRef, __construct)
+{
+	zval *referent;
+
+	if (zend_parse_parameters_throw(ZEND_NUM_ARGS(), "o", &referent) != SUCCESS) {
+		return;
+	}
+
+	zend_weakref_construct(getThis(), referent);
+}
+
+ZEND_METHOD(WeakRef, get)
+{
+	zend_weakref *wr;
+
+	if (zend_parse_parameters_none_throw() != SUCCESS) {
+		return;
+	}
+
+	wr = zend_weakref_fetch(getThis());
+
+	ZVAL_COPY(return_value, &wr->referent);
+}
+
+static const zend_function_entry zend_weakref_methods[] = {
+	ZEND_ME(WeakRef, __construct, zend_weakref_construct_arginfo, ZEND_ACC_PUBLIC)
+	ZEND_ME(WeakRef, get, zend_weakref_get_arginfo, ZEND_ACC_PUBLIC)
+	ZEND_FE_END
+};
+
+void zend_register_weakref_ce(void) /* {{{ */
+{
+	zend_class_entry ce;
+
+	INIT_CLASS_ENTRY(ce, "WeakReference", zend_weakref_methods);
+	zend_ce_weakref = zend_register_internal_class(&ce);
+	zend_ce_weakref->ce_flags |= ZEND_ACC_FINAL;
+
+	zend_ce_weakref->create_object = zend_weakref_new;
+	zend_ce_weakref->serialize = zend_class_serialize_deny;
+	zend_ce_weakref->unserialize = zend_class_unserialize_deny;
+
+	memcpy(&zend_weakref_handlers, zend_get_std_object_handlers(), sizeof(zend_object_handlers));
+	zend_weakref_handlers.offset = XtOffsetOf(zend_weakref, std);
+
+	zend_weakref_handlers.free_obj = zend_weakref_free;
+	zend_weakref_handlers.read_property = zend_weakref_no_read;
+	zend_weakref_handlers.write_property = zend_weakref_no_write;
+	zend_weakref_handlers.has_property = zend_weakref_no_isset;
+	zend_weakref_handlers.unset_property = zend_weakref_no_unset;
+	zend_weakref_handlers.get_property_ptr_ptr = zend_weakref_no_read_ptr;
+
+	zend_weakref_handlers.clone_obj = zend_weakref_clone;
+}
+/* }}} */
+
+
+/*
+ * Local variables:
+ * tab-width: 4
+ * c-basic-offset: 4
+ * indent-tabs-mode: t
+ * End:
+ * vim600: sw=4 ts=4 fdm=marker
+ * vim<600: sw=4 ts=4
+ */
```

Comment:
Hello, just one small nit I've missed earlier: These so called local variables for certain text editors have now been omitted in one of the latest commits, so they can be removed also. Thank you...